bedita-app/app_controller.php

```diff
@@ -1238,6 +1238,86 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
         $this->setSessionForObjectDetail($objects['items']);
     }
 
+    /**
+     * Change "lang" field in bulk to objects.
+     *
+     * Used by "list_objects_bulk.tpl" view element.
+     *
+     * @throws BeditaException
+     */
+    public function changeLanguageObjects()
+    {
+        if (!empty($this->params['form']['objects_selected'])) {
+            $objectsToModify = $this->params['form']['objects_selected'];
+            $this->changeBulkObjects($objectsToModify, 'lang', $this->data['lang']);
+        }
+    }
+
+    /**
+     * Change "rights" field in bulk to objects.
+     *
+     * Used by "list_objects_bulk.tpl" view element.
+     *
+     * @throws BeditaException
+     */
+    public function changeRightsObjects()
+    {
+        if (!empty($this->params['form']['objects_selected'])) {
+            $objectsToModify = $this->params['form']['objects_selected'];
+            $this->changeBulkObjects($objectsToModify, 'rights', $this->data['rights']);
+        }
+    }
+
+    /**
+     * Change objects field value in bulk.
+     *
+     * @param integer[] $objectIds - IDs of the objects to change
+     * @param string    $field     - name of the field to change
+     * @param string    $value     - new value for the field
+     *
+     * @throws BeditaException
+     */
+    protected function changeBulkObjects($objectIds, $field, $value)
+    {
+        $beObject = ClassRegistry::init('BEObject');
+        $this->Transaction->begin();
+
+        foreach ($objectIds as $id) {
+            $this->checkObjectWritePermission($id);
+
+            if ($beObject->isFixed($id)) {
+                throw new BeditaException(sprintf(__('Error: changing %s to a fixed object!', true), $field));
```

Comment:
Is this ok? Should I run gettext to generate pot files?
This text string has a variable in it, meaning that a translation is needed for all possible values of `$field`.

Reply:
yes, it's ok

Reply:
yes, you should relaunch `./cake.sh gettext update`